src/transformers/pipelines/fill_in_middle.py

```diff
@@ -0,0 +1,368 @@
+import enum
+import warnings
+
+from ..utils import add_end_docstrings, is_tf_available, is_torch_available
+from .base import PIPELINE_INIT_ARGS, Pipeline
+
+
+# Ensure availability of the required models
+if is_torch_available():
+    from ..models.auto.modeling_auto import MODEL_FOR_CAUSAL_LM_MAPPING_NAMES
+
+if is_tf_available():
+    from ..models.auto.modeling_tf_auto import TF_MODEL_FOR_CAUSAL_LM_MAPPING_NAMES
+
+
+class ReturnType(enum.Enum):
+    TENSORS = 0
+    NEW_TEXT = 1
+    FULL_TEXT = 2
+
+
+# List of models that support FIM
+SUPPORTED_MODELS = [
+    "bigcode/starcoder",
+    "codellama/CodeLlama-7b-hf",
+    "codellama/CodeLlama-13b-hf",
+    "codellama/CodeLlama-34b-hf",
+    "codellama/CodeLlama-7b-Python-hf",
+    "codellama/CodeLlama-13b-Python-hf",
+    "codellama/CodeLlama-34b-Python-hf",
+]
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class FIMPipeline(Pipeline):
```

Comment:
let's use camel casing
```suggestion
class FimPipeline(Pipeline):
```